tensorrt_llm/_torch/pyexecutor/resource_manager/base.py

```
@@ -0,0 +1,148 @@
+            # Clean up the state
+            for req in self.scheduled_requests.all_requests():
+                req.use_draft_model = False
+
+    return RequestContext(is_draft, scheduled_requests)
+
+
+class ResourceManager:
+    def __init__(self, resource_managers: dict[ResourceManagerType, BaseResourceManager]):
+        self.resource_managers = OrderedDict(resource_managers)
+
+    def __call__(self, type: ResourceManagerType):
```

Ruff (0.15.9)

[error] Function argument `type` is shadowing a Python builtin

(A002)